src/clarity-angular/wizard/wizard-page.spec.ts

```diff
@@ -39,31 +82,1201 @@ class TestComponent {
+
+                    <template pageHeaderActions>
+                        <clr-wizard-header-action (actionClicked)="headerActionClicked($event)" id="bell">
+                            <clr-icon shape="bell" class="has-badge"></clr-icon>
+                        </clr-wizard-header-action>
+                        <clr-wizard-header-action (actionClicked)="headerActionClicked($event)" id="warning">
+                            <clr-icon shape="warning"></clr-icon>
+                        </clr-wizard-header-action>
+                    </template>
+
+                    <template pageButtons>
+                        <clr-wizard-button [type]="'cancel'" class="clrtest-page-cancel"
+                            #pageCancelBtn>Cancel</clr-wizard-button>
+                        <clr-wizard-button [type]="'previous'"
+                            class="clrtest-page-previous-1">Previous</clr-wizard-button>
+                        <clr-wizard-button [type]="'danger'">Caution</clr-wizard-button>
+                    </template>
+                </clr-wizard-page>
+
+                <clr-wizard-page #viewTestWizardPageTwo
+                    [(clrWizardPagePreviousDisabled)]="disablePrevious">
+                    <template pageTitle>View Page 2</template>
+                    <p>{{projector}}</p>
+                </clr-wizard-page>
+
+                <clr-wizard-page #viewTestWizardPageThree
+                    [(clrWizardPagePreviousDisabled)]="disablePrevious">
+                    <template pageTitle>View Page 3</template>
+                    <template pageNavTitle>short title</template>
+                    <p *ngIf="!asyncLoaded">Loading...</p>
+                    <p *ngIf="asyncLoaded">{{asyncContent}}</p>
+
+                    <template pageButtons>
+                        <clr-wizard-button [type]="'cancel'">Cancel</clr-wizard-button>
+                        <clr-wizard-button [type]="'previous'" #pagePreviousBtn
+                            class="clrtest-page-previous-2">Previous</clr-wizard-button>
+                        <clr-wizard-button [type]="'danger'">Danger</clr-wizard-button>
+                    </template>
+                </clr-wizard-page>
+
+                <clr-wizard-page #viewTestWizardPageFour
+                    [clrWizardPagePreventDefaultCancel]="preventCancel"
+                    (clrWizardPageOnCancel)="altCancel()"
+                >
+                    <template pageTitle>View Page 4</template>
+                    <clr-alert [clrAlertClosable]="false">
+                        <div class="alert-item">
+                            <span class="alert-text">
+                                i believe the answer is {{innerProjector/2}}
+                            </span>
+                        </div>
+                    </clr-alert>
+                </clr-wizard-page>
+            </clr-wizard>
+            `
+        })
+        class ViewTestComponent {
```

Comment:
Not only does creating this monster of a component for each and every test seem like it would slow down everything and potentially create issue, you also _declare the whole class_ inside a Jasmine function. I have no idea how many times this is declared, how many times Angular compiles it on the fly, how many instances you get every time you compile the component after this, ... This should really, _really_ be at the module level.